icefoganalytics/elcc-data-management
# Dev Tooling: Automate the Migration and Seeding Process of the Database

Relates to https://github.com/icefoganalytics/elcc-data-management/issues/3

# Context

Currently initial database creation and the running of migrations and seeds are all manual processes. 
This PR automates database creation and the running of migrations and seeds.

## Questions

- [x] ~Should I update the db service "healthcheck" to check if the "ELCC" database is ready, rather than just checking if the sql server is ready?~ Yes.

# Implementation

1. Create a database initialization pipeline that waits for the database to spin up, then runs the SQL to create the database.
2. Run migrations on api service start.
3. Run seeds on api service start.

# Screenshots

Migrations and tables
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/ca6104fc-427a-4963-ac1d-bcdec29cae2d)

Seeds in "centers" table
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/ae818531-8b40-4d37-9a13-010612857a04)

# Testing Instructions

1. Run `dev down -v` to drop all services and databases for this project.
2. Run `dev up --build` to create the new database and run migrations and seeds.
3. Open your database manager and check that the appropriate tables exist and all migration have run.
5. Check that all seeds have run as well.
6. Open the app and check that you can log in.
